File: plugins/setup/src/main/java/net/elytrarace/setup/conversation/cup/CupPrompt.java
package net.elytrarace.setup.conversation.cup;

import net.elytrarace.api.conversation.BooleanPrompt;
import net.elytrarace.api.conversation.ConversationContext;
import net.elytrarace.api.conversation.Prompt;
import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class CupPrompt extends BooleanPrompt {

    @Override
    public @NotNull Component getPromptText(@NotNull ConversationContext context) {
        return Component.translatable("prompt.cup");
    }

    @Override
    protected @Nullable Prompt acceptValidatedInput(@NotNull ConversationContext context, boolean input) {
        if (input) {
            return new CupNamePrompt();
        }
        return END_OF_CONVERSATION;
    }
}

File: src/main/kotlin/net/elytrarace/conversation/cup/CupPrompt.kt
package net.elytrarace.conversation.cup

import net.elytrarace.conversation.BooleanPrompt
import net.elytrarace.conversation.ConversationContext
import net.elytrarace.conversation.Prompt
import net.kyori.adventure.text.Component


class CupPrompt : BooleanPrompt() {
    override fun getPromptText(context: ConversationContext): Component {
        return Component.translatable("prompt.cup")
    }

    override fun acceptValidatedInput(context: ConversationContext, input: Boolean): Prompt? {
        if (input) {
            return CupNamePrompt()
        }
        return END_OF_CONVERSATION
    }
}